improve Dockerfile build sequence

pifou25 opened the pull request:
## Description

Pour éviter de recopier les sources lorsque ce n'est pas nécessaire: le step (6) dans Dockerfile `COPY . ${WEBSERVER_HOME}` fait déjà la copie des fichiers dans le répertoire courant.
Pourtant je l'ai quand même laissé post-install dans le init.sh pour un cas particulier :

`docker run -p 80:80 -v /tmp/jeedom:/var/www/html --rm --name jeedom_server jeedom`
Avec l'option -v je map le rep de l'host sur celui du container. Du coup ça permet de persister Jeedom sur l'host, et ça marche lors de l'arrêt / relance du container, on ne perd rien. Mais au 1er lancement, si mon rep /tmp/jeedom est vide sur l'host, il est vide aussi dans le container, malgré que je pensais y avoir copié les sources (lors du build). Dans ce cas uniquement, on doit recopier les sources.
C'est ce que je ne comprends pas dans la doc, normalement si le rep est vide on devrait y retrouver les données du container:
https://docs.docker.com/engine/storage/volumes/#mounting-a-volume-over-existing-data

Le fichier `.dockerignore` permet à l'instruction `COPY . ` de savoir ce qu'il faut ignorer - ne pas copier dans l'image.

Le step 10 (install composer) se fait aussi au build de l'image, pas au démarrage de l'application. Donc dans le Dockerfile également.
J'ai mis à jour `composer.lock` de nouveau mais c'est normalement le même que celui en attente de merge sur #2905 mais le build ne marche pas avec le `composer.lock` actuel (c'est sans doute pour ça qu'il est systématiquement supprimé)

### Suggested changelog entry
<!-- Please provide a short description of the change for the changelog. -->


### Related issues/external references

Fixes #


## Types of changes
<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix _(non-breaking change which fixes)_
- [ ] New feature _(non-breaking change which adds functionality)_
- [ ] Breaking change _(fix or feature that would cause existing functionality to change)_
    - [ ] This change is only breaking for integrators, not for external standards or end-users.
- [ ] Documentation improvement


## PR checklist
<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
- [ ] I have checked there is no other PR open for the same change.
- [ ] I have read the [[La ligne directrice pour contribuer à ce projet / Contribution guidelines for this project](.github/CONTRIBUTING.md)).
- [ ] I grant the project the right to include and distribute the code under the GNU.
- [ ] I have added tests to cover my changes.
- [ ] I have verified that the code complies with the projects coding standards.
- [ ] [Required for new sniffs] I have added MD documentation for the sniff.

<!--
============================================================================================
Please make sure your pull request passes all continuous integration checks!

PRs which are failing their CI checks will likely be ignored by the maintainers.

PRs using atomic, descriptive commits are hugely appreciated as it will make
reviewing your changes easier for the maintainers.
============================================================================================
-->

Sekiro-kost committed:
feat: new maxsizeLog parameter, and log::chunck in cronHourly

Sekiro-kost committed:
fix variableName and re clear chunkLog

Sekiro-kost committed:
fix code

zoic21 committed:
Merge pull request #2930 from jeedom/alpha

merge

zoic21 committed:
Merge pull request #2931 from jeedom/alpha

merge

zoic21 committed:
Merge pull request #2934 from jeedom/alpha

beta

zoic21 committed:
Merge pull request #2936 from jeedom/alpha

beta

zoic21 committed:
Merge pull request #2938 from jeedom/alpha

merge

zoic21 committed:
Merge pull request #2940 from jeedom/alpha

Merge

zoic21 committed:
Merge pull request #2941 from jeedom/alpha

merge

zoic21 committed:
Merge pull request #2942 from jeedom/alpha

merge

zoic21 committed:
Merge pull request #2944 from jeedom/alpha

fix #2943

zoic21 committed:
Merge pull request #2945 from jeedom/alpha

bugfix

zoic21 committed:
Merge pull request #2946 from jeedom/alpha

prevent warning

zoic21 committed:
Merge pull request #2947 from jeedom/alpha

same as master

zoic21 committed:
Merge pull request #2948 from jeedom/alpha

typo

zoic21 committed:
Merge pull request #2949 from jeedom/alpha

test

zoic21 committed:
Merge pull request #2951 from jeedom/alpha

bugfix

zoic21 committed:
Merge pull request #2952 from jeedom/alpha

merge

zoic21 committed:
Merge pull request #2953 from jeedom/alpha

add text

zoic21 committed:
Merge pull request #2954 from jeedom/alpha

merge

zoic21 committed:
Merge pull request #2955 from jeedom/alpha

merge

zoic21 committed:
Merge pull request #2956 from jeedom/alpha

merge

zoic21 committed:
Merge pull request #2957 from jeedom/alpha

sync

zoic21 committed:
Merge pull request #2958 from jeedom/alpha

merge

zoic21 committed:
Merge pull request #2960 from jeedom/alpha

merge

zoic21 committed:
Merge pull request #2961 from jeedom/alpha

Update changelog.md

zoic21 committed:
Merge pull request #2965 from jeedom/alpha

beta

zoic21 committed:
Merge pull request #2971 from jeedom/alpha

merge

zoic21 committed:
Merge pull request #2975 from jeedom/alpha

stable

zoic21 committed:
Merge pull request #2977 from jeedom/alpha

beta

pifou25 committed:
improve Dockerfile build sequence

zoic21 committed:
Merge pull request #2996 from jeedom/alpha

merge to beta

Sekiro-kost committed:
Update administration.php

Hotfirenet commented on Dockerfile:
Il est un peu dommage de retirer l’utilisation de Composer de la procédure d’installation.
Y a-t-il une raison particulière à ce choix ?

pifou25 commented on Dockerfile:
Ha oui c'est la méthode préconisée dans un Dockerfile pour installer Composer, car on utilise directement le binaire d'une autre image docker (celle officielle de composer)
Mais on pourrait conserver le step 10 de install.sh qui fonctionne aussi :)

pifou25 commented on Dockerfile:
j'ai remis la procédure d'installation de composer, step 10.

Sekiro-kost committed:
add disabled on inputId

Sekiro-kost committed:
Update search.php

Sekiro-kost committed:
Update search.php

zoic21 committed:
Fix too much refresh

https://community.jeedom.com/t/refresh-de-la-tuile-sur-une-cmd-action-color/141380/6

Phpvarious committed:
Update user.class.php

ColonelMoutarde committed:
Create .editorconfig

chore(editorconfig) : nettoyage et simplification des règles

# .editorconfig
# Ce fichier définit des règles de style de code (indentation, encodage, fin de ligne, etc.)
# afin d'assurer une cohérence entre les différents éditeurs et IDEs utilisés par les développeurs.
# Il est automatiquement pris en charge par les éditeurs compatibles (comme VS Code, PHPStorm, etc.).
# Plus d'informations : https://editorconfig.org

- Simplification des motifs globaux (retrait des ** superflus)
- Ajout de insert_final_newline pour une meilleure cohérence des fichiers

Salvialf committed:
Merge pull request #3114 from jeedom/alpha

alpha to beta

Salvialf committed:
Merge pull request #3115 from jeedom/alpha

alpha to beta

zoic21 committed:
Merge pull request #3138 from jeedom/alpha

Merge

zoic21 committed:
Fix error Class "SolarData\SolarData" not found

Hotfirenet committed:
Merge pull request #3140 from jeedom/zoic21-patch-6

Fix error Class "SolarData\SolarData" not found

Salvialf committed:
update composer.lock

Salvialf committed:
Merge pull request #3141 from jeedom/composer-lock-update

update composer.lock

zoic21 committed:
Fix changelog with 4.4.20 version inplace of 4.4.19 for emergency restore cli

zoic21 committed:
Merge pull request #3142 from jeedom/zoic21-patch-6

Fix changelog with 4.4.20 version inplace of 4.4.19 for emergency res…

zoic21 committed:
Fix warning

https://community.jeedom.com/t/jeedom-4-5-une-version-plus-moderne-et-plus-fluide/144863/64

zoic21 committed:
Fix trigger name issue when it's a scenario

Sekiro-kost committed:
Merge pull request #3144 from jeedom/patch-2

zoic21 committed:
Fix 500 error on delete of some device

https://community.jeedom.com/t/probleme-lors-de-la-suppression-dun-equipement/144928

zoic21 committed:
Merge pull request #3143 from jeedom/zoic21-patch-7

Fix warning

franck-jeedom committed:
Update history.md

L'exportation CSV n'est pas dans l'écran principal, elle est disponible dans l'écran Configuration des Commandes.

zoic21 committed:
Merge pull request #3145 from jeedom/zoic21-patch-8

Fix 500 error on delete of some device

Salvialf committed:
init 4.5.1

Salvialf committed:
bump nodejs to v22

Salvialf committed:
Merge pull request #3148 from jeedom/3147-nodejs-22

bump nodejs to v22

Salvialf committed:
typo

Salvialf committed:
Merge pull request #3150 from jeedom/zoic21-patch-5

Fix too much refresh

Salvialf committed:
typo trigger_value

Salvialf committed:
typo changelog

Salvialf committed:
fix cd485a0

Salvialf committed:
code format

Salvialf committed:
$value must be a string

Salvialf committed:
improve color picker

TonioBDS committed:
Mise à jour du changelog avec dépréciations et clarifications

Ajout d'informations sur la dépréciation de la fonction `trigger()` et mise à jour de la description de `#trigger_name#` dans le changelog.

TonioBDS committed:
Update #trigger_name# tag explanation in scenario.md

Clarified the usage of the #trigger_name# tag and its implications for object, equipment, or command name changes.

zoic21 committed:
Try to ignore case and accent for function fromHumanReadable

zoic21 committed:
Fix random backup hour

Salvialf committed:
Merge pull request #3109 from Phpvarious/Phpvarious-user_connect

Update user.class.php

Salvialf committed:
update doc & changelog #3109

Salvialf committed:
Merge branch '4.5.1' of https://github.com/jeedom/core into 4.5.1

Salvialf committed:
Merge pull request #3156 from jeedom/3155-improve-color-picker-widget

improve color picker

Salvialf committed:
continue if value is null

zoic21 committed:
Merge pull request #3160 from jeedom/random-backup-hour

Fix random backup hour

zoic21 committed:
Merge pull request #3154 from jeedom/3152-php-deprecated-scenarioexpressionclass

3152 php deprecated scenarioexpressionclass

zoic21 committed:
Update changelog.md

Commit:
[Jenkins] Updated translation

zoic21 committed:
Merge pull request #3161 from jeedom/4.5.1

4.5.1

Salvialf committed:
init 4.5.2

Commit:
[Jenkins] Updated translation

Salvialf committed:
code format

Salvialf committed:
prevent warning

zoic21 committed:
Random backup only for cloud backup enable

zoic21 committed:
Update consistency.php

Salvialf committed:
fix 664d6b6

Salvialf committed:
remove jeeNetwork leftovers but keep system halt & reboot

noodom committed:
fix error in scenario search

Sekiro-kost committed:
Update market.display.repo.php

Sekiro-kost committed:
Update market.display.repo.php

zoic21 committed:
bugfix

BisonJeedom committed:
fix graphUpdate function in history class

Salvialf committed:
rebase 4.5.3 from master

Sekiro-kost committed:
Merge pull request #3174 from jeedom/fix-market-display-purchase

Update market.display.repo.php

zoic21 committed:
Merge pull request #3163 from jeedom/random-backup-cloud-only

Random backup only for cloud backup enable

zoic21 committed:
Correction bug envoi backup market avec un caractere specifique dans le login

zoic21 committed:
Correction bug envoi backup market avec un caractere specifique dans le login

Sekiro-kost committed:
Use floatval instead of intval for history values

Replaced intval with floatval when rounding historical average, min, and max values to ensure decimal precision is preserved. This change affects the calculation and display of historical statistics in the cmd class.

Sekiro-kost committed:
Merge pull request #3182 from Sekiro-kost/4.5.3

Use floatval instead of intval for history values

Salvialf committed:
code format

Salvialf committed:
improve cache persist

As same as #3184 without irrelevant documentation changes

Salvialf committed:
keep same method names as jeeApi

Salvialf committed:
Merge pull request #3168 from jeedom/3121-clean-jeenetwork-leftovers

remove jeeNetwork leftovers but keep system halt & reboot

Salvialf committed:
Merge branch 'bugfix-username-backupcloud' of https://github.com/jeedom/core into bugfix-username-backupcloud

Salvialf committed:
Merge pull request #3173 from noodom/patch-32

fix error in scenario search

Salvialf committed:
Merge pull request #3181 from jeedom/bugfix-username-backupcloud

Correction bug envoi backup market avec un caractere specifique dans le login

zoic21 committed:
Merge pull request #3162 from jeedom/ignore_accent_fromHumanReadable

Try to ignore case and accent for function fromHumanReadable

zoic21 committed:
Merge pull request #3157 from TonioBDS/patch-1

Mise à jour du changelog avec dépréciations et clarifications de trigger() et #trigger_name#

zoic21 committed:
Merge pull request #3146 from franck-jeedom/patch-1

Update history.md

zoic21 committed:
Merge pull request #3111 from ColonelMoutarde/patch-6

Create .editorconfig

zoic21 committed:
Update log.class.php

Salvialf committed:
update changelog

Salvialf committed:
Exclude system image files from install/update

zoic21 committed:
Merge pull request #3189 from jeedom/Typage-on-logadd

Update log.class.php

zoic21 committed:
Update core/class/jeedom.class.php

Co-authored-by: kwizer15 <kwizer15@users.noreply.github.com>

zoic21 committed:
Merge pull request #3045 from jeedom/2563-feat-fix-improve-log-rolling-purge

2563 feat fix improve log rolling purge

zoic21 committed:
Merge pull request #3095 from jeedom/disabled-searchEqlogicOnSearchModal

add disabled on input Equipment and Command, in searchModal

Salvialf committed:
Merge pull request #3192 from jeedom/3190-exclude-smartatlas-system-images-from-backups

Exclude `install/update/*.img.gz` from backup archive

zoic21 committed:
Update market.repo.php

Cloud backup auto clean itself no need to do that by jeedom

aofc committed:
Fix randText to evaluate string and return random value

Mips2648 committed:
fix: Merge setTags method to append tags instead of replacing them

zoic21 committed:
Revert "Try to ignore case and accent for function fromHumanReadable"

zoic21 committed:
Merge pull request #3201 from jeedom/revert-3162-ignore_accent_fromHumanReadable

Revert "Try to ignore case and accent for function fromHumanReadable"

zoic21 committed:
Update market.repo.php

Improve send market backup

Mips2648 committed:
Merge pull request #3198 from jeedom/fix-non-existent-tags

fix: Merge setTags method to append tags instead of replacing them

MrWaloo committed:
Suppression de la double suppression des NULL dans les tables d'historiques

Devant l'engouement suite au message sur le community qui a déchainé les foules et a reçu plein de réponses, je me suis senti obligé de faire cette PR.

Mips2648 committed:
Merge pull request #3202 from MrWaloo/history-delete-NULL

Suppression de la double suppression des NULL dans les tables d'historiques

Mips2648 committed:
Update Debian version check to support major versions 11 and 12

Mips2648 committed:
remove os version reference

Mips2648 committed:
Refactor: improve error handling in GitHub repo class

zoic21 committed:
Issue #3206

zoic21 committed:
typo

Mips2648 committed:
Merge pull request #3204 from jeedom/fix-github-repo

Refactor: improve error handling in GitHub repo class

zoic21 committed:
Merge pull request #3209 from jeedom/log-user-connection

Issue #3206 -  Add log on user connexion

Mips2648 committed:
rollback port rename

Salvialf committed:
Update OS support comment

Salvialf committed:
Merge pull request #3203 from jeedom/fix-debian-version-check

Fix debian version check & reference

Salvialf committed:
review gpio mapping names

Salvialf committed:
also list /dev/ttyAML*

Salvialf committed:
format code & fix potential type error

Salvialf committed:
add config searchValue static method

Salvialf committed:
auto update renamed ports in config

Salvialf committed:
update changelog

Salvialf committed:
precise searchValue description

Salvialf committed:
Merge pull request #3211 from jeedom/feat-review-usb-naming

Feat review usb naming

Salvialf committed:
add specific config for known boards

Salvialf committed:
update changelog

Salvialf committed:
update luna z2m controller

Salvialf committed:
not searchValue but byValue

Salvialf committed:
Merge pull request #3212 from jeedom/feat-review-usb-naming

add specific config for known boards

Salvialf committed:
hotfix prevent loops

Salvialf committed:
Merge pull request #3213 from jeedom/feat-review-usb-naming

hotfix prevent loops

Salvialf committed:
continue optimization

Salvialf committed:
further optimization

Salvialf committed:
Merge pull request #3214 from jeedom/feat-review-usb-naming

continue optimization

Salvialf committed:
format code

Salvialf committed:
fix 66b1273

prevent operand error from log::chunk called by jeedom::cronHourly (`Unsupported operand types: string * int`)

Salvialf committed:
Merge pull request #3215 from jeedom/feat-review-usb-naming

fix 66b1273

Salvialf committed:
format code

Salvialf committed:
skip random sleep when credentials are not configured

Avoids blocking cronHourly for up to 30 minutes when Market
credentials are empty.

Salvialf committed:
Merge pull request #3216 from jeedom/fix/market-skip-sleep-when-unconfigured

skip random sleep when credentials are not configured

Salvialf committed:
plugins handle hardware-specific config

Salvialf committed:
Merge pull request #3217 from jeedom/feat/plugin-specific-config

Handle hardware-specific configurations in plugins

zoic21 committed:
Merge pull request #3195 from jeedom/zoic21-patch-1

Update market.repo.php

Salvialf committed:
format code

Salvialf committed:
fix case

Salvialf committed:
additionnal dependancy check only when previously ok

Salvialf committed:
Merge pull request #3219 from jeedom/fix/dependancy_info

Fix/dependancy info

Salvialf committed:
add 4.5.3 details

Salvialf committed:
Merge pull request #3220 from jeedom/fix/changelog-add-details

add 4.5.3 changelog details

Phpvarious committed:
Update plugin.ajax.php

Add health into CommunityPost

Phpvarious committed:
Update plugin.ajax.php

Mips2648 committed:
Merge pull request #3224 from Phpvarious/patch-6

Update plugin.ajax.php

Salvialf committed:
update changelog

Salvialf committed:
typo

Salvialf committed:
Merge pull request #3230 from jeedom/fix/update-changelog

update changelog

Salvialf committed:
update changelog

Mips2648 committed:
fix: correct token handling in curl command for GitHub API requests

Salvialf committed:
Merge pull request #3232 from jeedom/fix/github-repo-with-token

fix: correct token handling in curl command for GitHub API requests

Salvialf committed:
Merge pull request #3231 from jeedom/4.5.3

4.5.3 stable release

Salvialf committed:
Merge pull request #3178 from BisonJeedom/master

fix graphUpdate function in history class

fgallnii committed:
fix: resolve 3 bugs in proxy configuration in jsonrpcClient

- Use strict comparison (== 1) for proxyEnabled to avoid truthy
  evaluation of unexpected default values
- Fix empty() receiving boolean from || operator instead of
  individual config values
- Replace hardcoded 'proxyLogin:proxyPassword' string with actual
  config::byKey() values
- Add proper error handling when proxy address or port is missing
- Remove log::add('Connection', 'debug', $ch) which logs a curl
  resource (not useful)

Mips2648 committed:
reduce if-else nesting

Co-authored-by: kwizer15 <kwizer15@users.noreply.github.com>

Mips2648 committed:
Fix random selection from empty array in scenarioExpression

Mips2648 committed:
Merge pull request #3238 from fgallnii/fix/proxy-jsonrpcClient-v2

Fix: resolve 3 bugs in proxy configuration in jsonrpcClient

Mips2648 committed:
Merge pull request #3197 from aofc/patch-9

Fix randText

Mips2648 committed:
fix: update GitHub workflows to use 'develop' branch instead of 'alpha' and 'beta'

Mips2648 committed:
Merge pull request #3240 from jeedom/fix/adapt-github-workflow-with-new-branches

chore: update GitHub workflows to use 'develop' branch instead of 'alpha' and 'beta'

Mips2648 committed:
typo Securité

Mips2648 committed:
Merge pull request #3241 from jeedom/fix/typo

typo Securité

Salvialf committed:
implode groupingType if is array

Salvialf committed:
Explicitly handle groupingType arrays with function/time keys

TiTidom-RC committed:
feat: add logProtect() support in log::remove()

Plugins can now declare a static logProtect(): array method to
prevent specific log files from being deleted by log::remove().
Protected logs are cleared (log::clear) instead of removed.

Pattern identical to backupExclude() in install/backup.php.
Fully backward-compatible: plugins without logProtect() are unaffected.

MrWaloo committed:
Correction *aucun* script

Mips2648 committed:
Merge pull request #3246 from MrWaloo/correction-aucun-script

Correction *aucun* script

TiTidom-RC committed:
Refactor log clearing to consider plugin daemons

Cache and use an active plugin list (excluding known core log names) to determine when to clear logs. Plugin IDs are sorted by length to avoid prefix collisions; if a log name starts with a plugin ID and refers to a sub-log while the plugin exposes a daemon, the sub-log is cleared. This also avoids plugins accidentally protecting core logs and reduces repeated plugin enumeration. (Includes a TODO to update the core log name list when new core logs are added.)

Mips2648 committed:
refactor: enhance log management methods with type hints and improved logic

zoic21 committed:
add auto translate to dev branch for test

zoic21 committed:
fix branch...

github-actions[bot] committed:
Auto update translation done by jeedom/jeetranslate workflow

zoic21 committed:
I'm rolling back because my commit doesn't follow the rules.

Mips2648 committed:
Merge pull request #3245 from TiTidom-RC/feat-protectlogs

feat: improve "do not remove log" feature

zoic21 committed:
Bugfix translation

Mips2648 committed:
fix: $actionScenario should always be a new instance to avoid corruption of current one

Mips2648 committed:
fix: ensure logging occurs before launching actionScenario with correct tags

Mips2648 committed:
refacto: streamline actionScenario launch for 'startsync' option

Mips2648 committed:
fix: update logging for actionScenario launch to differentiate between sync and regular scenarios

Salvialf committed:
Merge pull request #3242 from jeedom/fix/history-range-with-grouping

implode groupingType if is array

Mips2648 commented:
Hello,
si cette PR a encore un intérêt, il faudrait la re-soumettre sur develop car sinon elle embarque des changements non-désirés.

si plus besoin alors on peut la fermer

kwizer15 committed:
Migrer PHPStan vers une dépendance Composer

Mips2648 committed:
Merge pull request #3256 from kwizer15/feat/phpstan

Migrer PHPStan vers une dépendance Composer (edit #3110)

kwizer15 committed:
Suppression des arguments superflus dans les appels de méthodes (PHPStan lvl 1)

kwizer15 committed:
Correction des méthodes statiques et du nommage de classes (PHPStan lvl 1)

Salvialf committed:
fix display_name field (ui)

Mips2648 committed:
Merge pull request #3259 from kwizer15/fix/phpstan-static-methods

Correction des méthodes statiques et du nommage de classes (PHPStan lvl 1)

Mips2648 committed:
Merge pull request #3258 from kwizer15/fix/phpstan-extra-args

Suppression des arguments superflus dans les appels de méthodes (PHPStan lvl 1)

kwizer15 committed:
Corriger l'échec du step de suppression de branche dans le workflow PHPStan

git ls-remote --exit-code retourne 2 quand la branche n'existe pas,
ce qui, combiné à bash -e, faisait échouer tout le step. Remplacement
par une condition if/grep pour ne pas propager le code de sortie.

Mips2648 committed:
Merge pull request #3263 from kwizer15/fix/phpstan-workflow-branch-check

Corriger l'échec du workflow PHPStan lors de la suppression de branche

Mips2648 committed:
Merge pull request #3255 from jeedom/fix/scenario-self-execute-mixing-tags

Fix/scenario self execute mixing tags

Mips2648 committed:
Update PHPStan baseline

Mips2648 committed:
Merge pull request #3265 from jeedom/update-phpstan-baseline

[CI] Update PHPStan baseline

Salvialf committed:
Merge pull request #3262 from jeedom/fix/object-displayname-ui

fix display_name field (ui)

Salvialf committed:
Merge pull request #3251 from jeedom/feat/translation

Bugfix translation

kwizer15 committed:
Corriger l'injection SQL dans setComponentOrder et ajouter des tests d'intégration

Le champ `type` envoyé par le client dans `setComponentOrder` était concaténé
directement dans la requête SQL, sans aucune validation, contrairement aux
autres champs contrôlés par is_numeric(). Un admin pouvait injecter du SQL
arbitraire (testé : `DROP TABLE viewData` s'exécute réellement).

- Whitelist stricte sur le champ `type` (cmd, eqLogic, scenario)
- Remplacement de la concaténation SQL multi-statements par un prepared
  statement exécuté à chaque itération
- Tests d'intégration couvrant l'endpoint via un sous-processus PHP
  authentifié comme admin (payload DROP, UNION, type inconnu, type manquant)

Mips2648 committed:
Merge pull request #3267 from kwizer15/fix/view-sql-injection

fix: prevent a possible SQL injection in setComponentOrder

pifou25 commented:
Je n'arrive pas à changer de branche sur celle-ci, alors j'ai fait une nouvelle PR https://github.com/jeedom/core/pull/3274
